src/xma/src/xmaapi/xmahw_hal.cpp

```diff
@@ -123,159 +124,72 @@ bool hal_configure(XmaHwCfg *hwcfg, XmaXclbinParameter *devXclbins, int32_t num_
+        for (int32_t d = 0; d < info.ip_axlf->m_count; d++) {
+            if (info.ip_axlf->m_ip_data[d].m_type != IP_KERNEL)
+                continue;
+            auto kernel_name = std::string((char*)info.ip_axlf->m_ip_data[d].m_name);
             dev_tmp1.kernels.emplace_back(XmaHwKernel{});
             XmaHwKernel& tmp1 = dev_tmp1.kernels.back();
             memset((void*)tmp1.name, 0x0, MAX_KERNEL_NAME);
-            info.ip_layout[d].kernel_name.copy((char*)tmp1.name, MAX_KERNEL_NAME-1);
-
-            tmp1.base_address = info.ip_layout[d].base_addr;
+            kernel_name.copy((char*)tmp1.name, MAX_KERNEL_NAME - 1);
             tmp1.cu_index = (int32_t)d;
-            if (info.ip_layout[d].soft_kernel) {
-                tmp1.soft_kernel = true;
-                tmp1.default_ddr_bank = 0;
-            } else {
-                tmp1.arg_start = info.ip_layout[d].arg_start;
-                tmp1.regmap_size = info.ip_layout[d].regmap_size;
-
-                if (info.ip_layout[d].kernel_channels) {
-                    tmp1.kernel_channels = true;
-                    tmp1.max_channel_id = info.ip_layout[d].max_channel_id;
-                }
-                //Allow default ddr_bank of -1; When CU is not connected to any ddr
-                xma_xclbin_map2ddr(info.ip_ddr_mapping[d], &tmp1.default_ddr_bank, info.has_mem_groups);
-
-                //XMA now supports multiple DDR Banks per Kernel
-                tmp1.ip_ddr_mapping = info.ip_ddr_mapping[d];
-                for(uint32_t c = 0; c < info.number_of_connections; c++)
-                {
-                    auto& xma_conn = info.connectivity[c];
-                    if (xma_conn.m_ip_layout_index == (int32_t)d) {
-                        tmp1.CU_arg_to_mem_info.emplace(xma_conn.arg_index, xma_conn.mem_data_index);
-                        //Assume that this mem is definetly in use
-                        if ((uint32_t)xma_conn.mem_data_index < dev_tmp1.number_of_mem_banks && xma_conn.mem_data_index > 0) {
-                            dev_tmp1.ddrs[xma_conn.mem_data_index].in_use = true;
-                        }
-                    }
-                }
-
-                if (tmp1.default_ddr_bank < 0) {
-                    xma_logmsg(XMA_WARNING_LOG, XMAAPI_MOD,"\tCU# %d - %s - default DDR bank: NONE", d, (char*)tmp1.name);
-                } else {
-                    xma_logmsg(XMA_DEBUG_LOG, XMAAPI_MOD,"\tCU# %d - %s - default DDR bank:%d", d, (char*)tmp1.name, tmp1.default_ddr_bank);
+            //Allow default ddr_bank of -1; When CU is not connected to any ddr
+            // TODO: Removed this code and handle it using xrt::kernel
+            xma_xclbin_map2ddr(info.ip_ddr_mapping[d], &tmp1.default_ddr_bank, info.has_mem_groups);
+            //XMA now supports multiple DDR Banks per Kernel
+            tmp1.ip_ddr_mapping = info.ip_ddr_mapping[d]; 
+            for (int32_t c = 0; c < info.conn_axlf->m_count; c++)
```

Comment:
mem_topo is also part of xrt::xclbin, can it be used?

Reply:
addressed this comment and used XRT::XCLBIN class in the code.